lib/testcheck.js

```diff
@@ -21,22 +21,40 @@
 
 "use strict";
 
-module.exports = function (sha, dbHost) {
+module.exports = function (sha, dbHost, maxnumberoftries, delay) {
 
     var http = require("http"),
         url = require("url"),
         q = require("q");
 
+    function tryConnect(options, d, pendingnumberoftries, operation, successCallBack){
+        http.get(options, successCallBack)
+            .on("error", function(e){
+                if(pendingnumberoftries > 1){
+                    console.log("Connection attempt to " + operation + " failed. Will try after: " + delay + " milliseconds.");
```

Comment:
Should you be using the util log function here - as it has some nice functionality to put time etc?